### PR TITLE
HBW-148 Do not submit empty values

### PR DESCRIPTION
**Pull request by sugarfree1**

### CODE REVIEW

File: hbw/app/models/hbw/form.rb [hunk 1]
```diff
@@ -58,5 +58,9 @@ def flatten_fields
         field.type == :group ? field.fields : field
       end
     end
+
+    def fields_for_submit(values)
+      flatten_fields.select(&:has_value?).select(&:editable?).select { |f| values.keys.include?(f.code) }
```
Comment (TimAle) [inline hbw/app/models/hbw/form.rb]: `.select(&:has_value?).select(&:editable?)` is duplicated.

**Comment by sugarfree1** [comment]
Updated

**Comment by sugarfree1** [comment]
Updated

File: hbw/config/yml_api.test.camunda.yml [hunk 2]
```diff
@@ -1372,16 +1372,17 @@ post:
...
+    "variables[homsOrderDataAddress][value]=test+address&variables[homsOrderDataBeginDate][value]=2016-09-30T11:52:20+00:00&variables[homsOrderDataEndDate][value]=2016-09-30T11:52:20+00:00&variables[homsOrderDataHomePlace][value]=test+home+place&variables[homsOrderDataRecommendation][value]=&variables[homsOrderDataSelect][value]=123456&variables[homsOrderDataSomeCheckbox][value]=false&variables[homsOrderDataSomeDatetime][value]=&variables[homsOrderDataSomeDisabledCheckbox][value]=false&variables[homsOrderDataSomeDisabledSelect][value]=&variables[homsOrderDataSomeDisabledText][value]=&variables[homsOrderDataSomeSelect][value]=&variables[homsOrderDataSomeStr][value]=&variables[homsOrderDataSomeText][value]=&variables[homsOrderDataWorkAddress][value]=&variables[homsOrderDataWorkPlace][value]=":
+    "variables[homsOrderDataAddress][value]=test+address&variables[homsOrderDataBeginDate][value]=2016-09-30T11:52:20+00:00&variables[homsOrderDataEndDate][value]=2016-09-30T11:52:20+00:00&variables[homsOrderDataHomePlace][value]=test+home+place&variables[homsOrderDataRecommendation][value]=&variables[homsOrderDataSelect][value]=123456":
```
Comment (TimAle) [inline hbw/config/yml_api.test.camunda.yml]: It was one line and now it is not.

**Pull request merged**